src/main/java/bridge/domain/BridgeGame.java

```diff
@@ -0,0 +1,71 @@
+package bridge.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * 다리 건너기 게임을 관리하는 클래스
+ */
+public class BridgeGame {
+
+    private final List<BridgeToken> bridgeTokens = new ArrayList<>();
+    private final Bridge bridge;
+    private int retryCount = 1;
+
+    public BridgeGame(List<String> bridge) {
+        this.bridge = new Bridge(bridge);
+    }
+
+    /**
+     * 사용자가 칸을 이동할 때 사용하는 메서드
+     * <p>
+     * 이동을 위해 필요한 메서드의 반환 타입(return type), 인자(parameter)는 자유롭게 추가하거나 변경할 수 있다.
+     */
+    public boolean move(Direction direction) {
+        if (tryForward(direction)) {
+            bridgeTokens.add(new BridgeToken(direction, true));
+            bridge.moveForward();
+            return true;
+        }
+        bridgeTokens.add(new BridgeToken(direction, false));
+        return false;
+    }
+
+    /**
+     * 사용자가 게임을 다시 시도할 때 사용하는 메서드
+     * <p>
+     * 재시작을 위해 필요한 메서드의 반환 타입(return type), 인자(parameter)는 자유롭게 추가하거나 변경할 수 있다.
+     */
+    public boolean retry(GameCommand gameCommand) {
+        if (gameCommand.equals(GameCommand.RETRY)) {
+            initGame();
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
```suggestion
    public BoardGame retry(GameCommand gameCommand) {
        if (gameCommand.equals(GameCommand.RETRY)) {
            return new BridgeGame(bridge);
        }
        return this;
    }
```

이전 코멘트와 비슷합니다. 이런 말이 있죠, `묻지 말고 시켜라!`.  retry 가 호출되면 boolean 값에 신경쓰지 않고 다음 동작을 계속 수행할 수 있도록 설계해보시면 좋을 것 같네요. 현재는 포지션 개념을 `Bridge` 가 들고 있기 때문에 제안된 코드는 제대로 동작하지 않을거라서 관련 부분을 어떻게 처리할지 추가적으로 고민해보시는 것도 재밌을 것 같네요.